cfme/tests/physical_infrastructure/test_redfish_physical_server_buttons.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+import pytest
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.common.physical_server_views import (
+    PhysicalServerDetailsView,
+)
+from cfme.physical.provider.redfish import RedfishProvider
+
+pytestmark = [pytest.mark.provider([RedfishProvider], scope="function")]
+
+
+@pytest.fixture(scope="function")
+def physical_server(appliance, provider, setup_provider_funcscope):
+    """Get and return the first physical server."""
+    yield appliance.collections.redfish_physical_servers.all(provider)[0]
+
+
+def assert_message(physical_server, provider, expected_message):
+    """
+    Assert that the physical server details view displays the requested message.
+
+    Args:
+      physical_server: check the details view of this physical server.
+      provider: the RedfishProvider instance
+      expected_message: the message we expect to be displayed in the flash alert
+    """
+    view = provider.create_view(PhysicalServerDetailsView, physical_server)
+    view.flash.assert_message(expected_message)
+
+
+# Power Button
+def test_redfish_server_power_off(physical_server, provider):
+    physical_server.power_off()
+    assert_message(physical_server, provider, "Requested power_off of selected item.")
+
+
+def test_redfish_server_power_on(physical_server, provider):
+    physical_server.power_on()
+    assert_message(physical_server, provider, "Requested power_on of selected item.")
+
+
+def test_redfish_server_power_off_immediately(physical_server, provider):
+    physical_server.power_off_immediately()
+    assert_message(physical_server, provider, "Requested power_off_now of selected item.")
+
+
+def test_redfish_server_restart(physical_server, provider):
+    physical_server.restart()
+    assert_message(physical_server, provider, "Requested restart of selected item.")
+
+
+def test_redfish_server_restart_immediately(physical_server, provider):
+    physical_server.restart_immediately()
+    assert_message(physical_server, provider, "Requested restart_now of selected item.")
+
+
+# Identify Button
+def test_redfish_server_turn_on_led(physical_server, provider):
+    physical_server.turn_on_led()
+    assert_message(physical_server, provider, "Requested turn_on_loc_led of selected item.")
+
+
+def test_redfish_server_turn_off_led(physical_server, provider):
+    physical_server.turn_off_led()
+    assert_message(physical_server, provider, "Requested turn_off_loc_led of selected item.")
+
+
+def test_redfish_server_turn_blink_led(physical_server, provider):
+    physical_server.turn_blink_led()
+    assert_message(physical_server, provider, "Requested blink_loc_led of selected item.")
+
+
+# Lifecycle Button
+def test_redfish_server_lifecycle_provision(physical_server):
+    view = navigate_to(physical_server, "Provision")
+    assert view.is_displayed
```

Comment:
Did we forget to press the button here or is it hidden in `navigate_to`?

Reply:
This does not test the button pressing functionality (which apparently is not an action but just a link to the "Provision" page, so we just do it directly.

Reply:
IMAO we needn't this test because I assume you'd soon be implementing a more complete test that will actually perform provisioning; and it'll click on Lifecycle > Provision Physical Server button anyway. That being said, if you want to keep it here, please add a docstring that we're just testing whether button redirects us to the desired page.

Reply:
👍